### PR TITLE
Remove LSBCheck

### PR DESCRIPTION
**Pull request by kstreitova**
Separate the check() method to multiple functions and clean up the code. Also add a test for LSBCheck and adjust LSBCheck.toml description to be more precise (#225 #156).

### CODE REVIEW

File: rpmlint/checks/LSBCheck.py [hunk 1]
```diff
@@ -18,15 +9,50 @@
```
Comment (scarabeusiv) [inline rpmlint/checks/LSBCheck.py]: The regexps could be class variable maybe, but regardless the a-zA-Z0-9... should be \w+ see the https://docs.python.org/3/howto/regex.html

**Comment by kstreitova** [comment]
After a discussion with Tomas, I suggest removing this check. See reasons in my commit message.

**Comment by scarabeusiv** [comment]
It is not enabled on any distro so LGTM. @Conan-Kudo your opinion?
Today it seems we delete shitaton of stuff ;-)

**Comment by Conan-Kudo** [comment]
I can explain where the rules come from. The rules are the intersection of compatible NVRs between RPM v3 and DPKG for supporting portability across RPM and Debian systems through tools like `alien` and other things.

The check is disabled by default in Fedora because we have stopped enforcing this compatibility requirement, but there are corporate users that have this check turned on for their own RPMs.

**Comment by kstreitova** [comment]
Then we should rename it as it's not connected to LSB at all. Any suggestions?

Are the regexps correct so I can adjust description according to it?

**Comment by Conan-Kudo** [comment]
@kstreitova It is related to LSB, as LSB specified RPM v3 as the standardized package format.

**Pull request closed without merging**